app/lib/search/backend.dart

```diff
@@ -51,13 +51,16 @@ class SearchBackend {
 
   /// Loads the list of packages, their latest stable versions and returns a
   /// matching list of [PackageDocument] objects for search.
-  /// When a package, its latest version or its analysis is missing, the method
-  /// returns with null at the given index.
+  ///
```

Comment:
I'd rather keep the return of null values. It follows the same consistency model as `DatastoreDB.lookup`: if you pass in 10 items, the result for the 3rd will be always in the given index, and you don't need to check if it is the same position as you've requested.

I'd prefer adding another flag to `PackageDocument` and post-filtering on it when needed.